### PR TITLE
Grenoble workshop summary

### PR DESCRIPTION
**Pull request by jouvin**
Obivously not ready for merging yet!

### CODE REVIEW

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+  * This should be implemented after the compilation and before deployment, looking at affected machines: may be doable a the broker level, not
+  requiring compiler changes
+  * But this would require some sort of host grouping and ownership in Aquilon
+  * Difficult in SCDB as no real authentication against the deployment server: only SVN transaction is authenticated
+  
+
+## CCM
+
+JSON and types: JSON has untyped data which is a problem when building configuration files where the data should be types
+* Stijn added a the ability to do a "typed interpretation" of JSON data (open PR): giving the same results as an XML profiles for unit tests
```
Comment (stdweird) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: the code is already merged, but the code is not used anywhere (or can't be in any way, eg.g via config file)

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+* Agreement to put it disabled in 15.2 and enabled by default in 15.4 after testing by sites
+
+ncm-query: need to support more output formats (like JSON) and add unit tests for output formats
+* Proposal
+  * move output formatting to a CCM module (CCMFormatter?)
+  * Move ncm-query to CCM repo
+
+ncm-query tab completion: current implementation is killed by the time to get the information from the database
+* Proposal: cache all paths present in the profile in a text file and use this file for time completion
+  * Could be done transparently fy `CCM/Fetch.pm`
```
Comment (stdweird) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: `fy` -> `by`

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+TT Toolkit template are very powerfull and can replace perl codes... but they may also be a source of bugs! Thus the need for unit tests
+* A test framework developed: `Test::Quattor::RegexpTest`
+  * Part of the test framework provided in `maven-tools`
+* Allow regexp pattern matching on `TextRender` output: ordered list of regexp can be defined
+  * By default, regexp are compiled multilines
+  * Possibility to negate regexps, either all of thems or a single one (COUNT statement)
+  
+`ncm-metaconfig`: high level interface to `TextRender`
+* Files managed by `metaconfig` are called `services`
+  * File name can now be defined in a global variable: see examples
```
Comment (stdweird) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: well, only the bind works. until also assigment is possible, i'm not sure how to promote this sort of usage.

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+* Can bind the contents to a schema
+* Can restart a daemon (or a list of daemons) in case of changes using `CAF::Service`
+  * When using the list of daemons (`daemons` property), an action can be specified on the daemon
```
Comment (stdweird) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: it already supports this, i would not mention the `daemon` field anywhere anymore

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+## Round table - Expectations
+
+* Release: solve the issues with release process
+* RH 7 support
+* Information on how to use Quattor more efficiently
+
+## Site news
+
+RAL: Aquilon is now in production
+* Private cloud, including a VM provisionning prototype service
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: provisioning

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+* Focus on support finer granularity changes: main challenge is allowing non expert user to use Aquilon/Quattor
+* Quattor Remote Deployer: still used for virtualization but no manpower to implement the necessary improvements
+
+
+## Platform Support
+
+UGent manages RHEL7 machines since last summer. Several minor issues with pull requests open:
+- AII Kickstart config
+  - KS variant available but misnamed `rhel7rc` instead of `el7`
+- LVM: several commands require --force flag
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: \`--force\`

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+Major issue: `ncm-chkconfig`, started to work `ncm-systemd`
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: started work on?

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+* Main issue is to get the mapping between a grub entry value and a kernel version: no trivial match between kernel version in Quattor config and
+the kernel title in Grub conf
+
+UGent deadline for solving all issues: June
+
+MS reports problems with `ncm-pam`: should open an issue
+
+## ncm-spma new approach by MS - J. Novy
+
+A different `yum.pm` to ensure convergence to the defined state in one run, like it was the case wit `SPMA`
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: with

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+* MS runs `ncm-spma` only at boot time, so very rarely
+* All packages not need must be removed at the first run
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: needed

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+* Performance similar to current `yum.pm` implementation
+* Also use YUM `FastestMirror` plugin
+
+Complete rewrite of `yum.pm`: need to be exposed on GitHub for further discussions
+
+
+## Release process and current release status
+
+14.12 didn't come out because of the changes in the way unit tests are run during the build process
+* Before: RPMs were built, unit tests were run and then release was tagged
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: Before: Unit tests were run as a separate step, then RPMs were built and release was tagged.

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+* Now: unit tests are required to run successfully for the release being tagged
+  * Required to build templates for metaconfig modules
+  * In principle a good change: unit tests have to succeed on all supported platforms or they are not useful.
+  
+Supported platforms at this time: EL5, EL6, EL7
+* Solaris: should add it as a supported platform starting with 15.4
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: Solaris 11

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+  * Do not build packages but run the unit tests as part of the release process
+
+There is also the issue of providing versions more recent than EPEL for some package in Quattor externals: general agreement that is should be avoided
+as much as possible
+* Providing our own version of a package create an obligation to maintain it
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: creates

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+* Open question: TTConfig. Version in EPEL doesn't work for some metaconfig services, need to see if we can work around.
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: s/TTConfig/template toolkit/

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 14]
```diff
@@ -0,0 +1,306 @@
...
+* Move these services to next release to unblock the release process
+* Still the problem of EL5: more tests failing...
+  * Release without EL5 tests: check and list problematic services on EL5, acceptable to have metaconfig services unsupported in EL5
+  
+Nexus central repo: need to ensure that enough people have the right to do new releases of build tools
+* Michel to check
+
+### Quattor externals
+
+Agreement to replace existing `externals` repo on `yum.quattor.org` by the new structure`currently under `.externals
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: the new structure currently under `.externals`

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 15]
```diff
@@ -0,0 +1,306 @@
...
+* No real need for restricting changes to some part of the configuration, rather to limit changes to some machines
+  * This should be implemented after the compilation and before deployment, looking at affected machines: may be doable a the broker level, not
+  requiring compiler changes
+  * But this would require some sort of host grouping and ownership in Aquilon
+  * Difficult in SCDB as no real authentication against the deployment server: only SVN transaction is authenticated
+  
+
+## CCM
+
+JSON and types: JSON has untyped data which is a problem when building configuration files where the data should be types
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: JSON doesn't have untyped data, but the way we are reading it into Perl with JSON::XS means we are losing the type information.

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+* `includepath` currently hardcoded and restricted to one value
+* Today restricted to developing services that are shared with the community: good side, will improve the quality by exposing the code! But need to see
+how to allow site-specific services
+  * Will require at least the ability to ship TT files independently of `ncm-metaconfig`: this also implies the required support in maven tools...
+  * This must maintain the ability to unit test things
+  * More discussion needed: see https://github.com/quattor/configuration-modules-core/issues/436
+  * James: must not lead to less site sharing and a bigger mess...
+  
+Issue to be addressed with the building process of TT files: currently done in the `test` phase which is not intended to produce anything required to
+build the package: need to enhance the build toools to allow doing it in the  `compile` phase.
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: no ooo in "tools"

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+* See https://github.com/quattor/maven-tools/issues/40
+
+### First candidates for ncm-metaconfig replacement
+
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: See https://github.com/quattor/configuration-modules-core/issues/434

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+* ncm-pnp4nagios
+* ncm-postfix: already using TT files
+* ncm-sysconfig: prologue/epilogue handling could be a problem
+
+Deprecated: ncm-fsprobe
+
+Need to do the same review for grid components.
+* Start with the components still relying on ncm-templates: use `CAF::TextRender` when a component remains needed (e.g. `glitestartup`)
+
+Some components may required the ability to configure path where part of the path is a variable
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: require

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 19]
```diff
@@ -0,0 +1,306 @@
...
+* ncm-openldap
+* ncm-openvpn
+* ncm-pnp4nagios
+* ncm-postfix: already using TT files
+* ncm-sysconfig: prologue/epilogue handling could be a problem
+
+Deprecated: ncm-fsprobe
+
+Need to do the same review for grid components.
+* Start with the components still relying on ncm-templates: use `CAF::TextRender` when a component remains needed (e.g. `glitestartup`)
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: where the need for a dedicated component remains

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+* 14.10: first attempt at using readthedocs.org
+
+Documentation source: currently a mix of inline documentation for Perl modules and separate pod files for the config modules schema
+* Config modules schema could be moved to annotations but will require a tool to process pan annotations to generate a documentation file
+  * Review Nick's tool to see if it is a good basis: see https://github.com/quattor/release/issues/46
+  * Michel: will attempt to annotate one schema file to help agreeing on the tags
+* Config modules pod files
+  * Broken link and reference to bugs, dependencies
+  * Hard coded references to people / email addresses
+  * Special characters are often not properly escaped wit `C<< >>`
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: with

**Comment by jouvin** [comment]
Shoud be ready for merging... would be good to distribute it to the community asap...

File: _posts/2015-03-03-grenoble-workshop-summary.md [hunk 21]
```diff
@@ -0,0 +1,293 @@
...
+  * This should be implemented after the compilation and before deployment, looking at affected machines: may be doable a the broker level, not
+  requiring compiler changes
+  * But this would require some sort of host grouping and ownership in Aquilon
+  * Difficult in SCDB as no real authentication against the deployment server: only SVN transaction is authenticated
+  
+
+## CCM
+
+JSON and types: the way we are reading JSON profiles (`JSON::XS`) results in type information being lost
+* A problem when building configuration files where the data should be types
```
Comment (jrha) [inline _posts/2015-03-03-grenoble-workshop-summary.md]: typed

**Comment by jrha** [comment]
I agree, one last fix and then I'll merge it.

**Pull request merged**